### PR TITLE
CA1849 Analyzer Documentation

### PR DESCRIPTION
**Pull request by mahdiva**
This analyzer is still in PR: https://github.com/dotnet/roslyn-analyzers/pull/5367

Opening this PR for now until the analyzer is merged in

### CODE REVIEW

**Comment by Youssef1313** [comment]
@gewarren The product PR was merged. This should be ready for review.

File: docs/fundamentals/code-analysis/quality-rules/ca1849.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+| | Value |
+|-|-|
+| **Rule ID** |CA1849|
+| **Category** |[Performance](performance-warnings.md)|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+All methods where an Async-suffixed equivalent exists will produce this warning when called from a Task-returning method. In addition, calling `Task.Wait()`, `Task<T>.Result` or `Task.GetAwaiter().GetResult()` will produce this warning.
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca1849.md:24]:
```suggestion
All methods where an Async-suffixed equivalent exists will produce this warning when called from a Task-returning method. In addition, calling `Task.Wait()`, `Task<T>.Result`, or `Task.GetAwaiter().GetResult()` will produce this warning.
```

File: docs/fundamentals/code-analysis/quality-rules/ca1849.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+```csharp
+async Task DoAsync()
+{
+    await file.ReadAsync(buffer, 0, 10);
+}
+```
+
+## When to suppress warnings
+
+It's safe to suppress a warning from this rule in the case where there are two separate code paths for sync and async code, using an if condition. Also if there is a check for whether the Task has resolved, it is safe to use sync methods and properties.
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca1849.md:54]:
```suggestion
It's safe to suppress a warning from this rule in the case where there are two separate code paths for sync and async code, using an if condition. Also if there is a check for whether the Task has resolved, it is safe to use sync methods and properties.
```

File: docs/fundamentals/code-analysis/quality-rules/ca1849.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+}
+```
+
+## When to suppress warnings
+
+It's safe to suppress a warning from this rule in the case where there are two separate code paths for sync and async code, using an if condition. Also if there is a check for whether the Task has resolved, it is safe to use sync methods and properties. 
+
+## See also
+
+- [Performance rules](performance-warnings.md)
```
Comment (BillWagner) [inline docs/fundamentals/code-analysis/quality-rules/ca1849.md:58]:
```suggestion
- [Performance rules](performance-warnings.md)

```

**Review by BillWagner — APPROVED** [review]
These changes LGTM, and the reference analyzer PR has been merged.

I'll :shipit: now.

**Pull request merged**